### PR TITLE
#679 Ask for confirmation before "Mark all as Read"

### PR DESCRIPTION
**Pull request by bogdanteodoru**

### CODE REVIEW

**Comment by bogdanteodoru** [comment]
Also, https://github.com/flarum/flarum-ext-english/pull/59 needs to be merged as well for this.

**Comment by franzliedke** [comment]
Hmm, I believe #679 is not about the notification list, but rather the little tick icon in the discussion list...,

**Comment by bogdanteodoru** [comment]
Ah. Sorry. Should I  keep this implementation as well or remove it and make it there?

**Comment by bogdanteodoru** [comment]
I have updated the pull request to reflect task changes.

**Comment by sijad** [comment]
@bogdanteodoru I think you need to make some changes to prevent password confirmation for mark all as read

**Comment by bogdanteodoru** [comment]
@sijad On my local environment there is not password asked when I mark them as read. Do I miss something here?

**Comment by sijad** [comment]
@bogdanteodoru I guess you didn't install latest dev version, please follow [this instruction](https://github.com/flarum/flarum/blob/master/CONTRIBUTING.md#development-setup), *sudo mode* isn't available in beta4, see 9896378b59b3f1d260049c162abe5786c4be8ed0

**Comment by bogdanteodoru** [comment]
Ok. I'm trying to install the dev-master for core and I'm getting this:

screen shot 2016-01-10 at 15 23 53

screen shot 2016-01-10 at 15 23 48

Any idea what I'm doing wrong? The documentation is rather ambiguous.

**Comment by sijad** [comment]
change `dev-master`s to `@dev` it should be like this:
```
"flarum/composer-installer": "*",
"flarum/core": "@dev",
"flarum/flarum-ext-akismet": "@dev",
"flarum/flarum-ext-approval": "@dev",
...
```
I'll open an issue for this

File: js/forum/src/components/NotificationList.js [hunk 1]
```diff
@@ -68,36 +68,36 @@ export default class NotificationList extends Component {
...
+
+                <ul className="NotificationGroup-content">
+                  {group.notifications.map(notification => {
+                    const NotificationComponent = app.notificationComponents[notification.contentType()];
+                    return NotificationComponent ? <li>{NotificationComponent.component({notification})}</li> : '';
+                  })}
+                </ul>
+              </div>
+            );
+          })
```
Comment (franzliedke) [inline js/forum/src/components/NotificationList.js]: Looks like the indentation is now wrong.

File: js/forum/src/components/DiscussionListItem.js [hunk 2]
```diff
@@ -174,8 +174,12 @@ export default class DiscussionListItem extends Component {
     const discussion = this.props.discussion;
 
     if (discussion.isUnread()) {
-      discussion.save({readNumber: discussion.lastPostNumber()});
-      m.redraw();
+      const confirmation = confirm(app.translator.trans('core.forum.discussion_list.mark_all_as_read_confirmation'));
+
+      if(confirmation) {
```
Comment (franzliedke) [inline js/forum/src/components/DiscussionListItem.js]: You're missing a space after the "if" here. :)

**Comment by franzliedke** [comment]
@bogdanteodoru We're almost there now... :)

Please fix the remaining issues and squash your commits into one, then I can merge.

**Comment by bogdanteodoru** [comment]
@franzliedke Yeah almost there. Sorry for all the mess but until I setup the project right on my workspace there might be some wrong bits and pieces here and there.

**Comment by bogdanteodoru** [comment]
@sijad the @dev method works properly. Thanks.

**Pull request closed without merging**